### PR TITLE
Populate SystemProperties for SVM runtime

### PR DESCRIPTION
**Pull request by Karm**
Certain monitoring tools that rely on sys props, e.g. [Micrometer](https://github.com/micrometer-metrics/micrometer/blob/main/micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmInfoMetrics.java), might report unpopulated fields when operating with native-image built apps, e.g.

[Quarkus Micrometer Quickstart](https://github.com/quarkusio/quarkus-quickstarts/tree/main/micrometer-quickstart) reports in native mode:

```
jvm_info_total{runtime="unknown",vendor="Oracle Corporation",version="unknown",} 1.0
```

while it works correctly in HotSpot, JVM mode:

```
jvm_info_total{runtime="OpenJDK Runtime Environment",vendor="Eclipse Adoptium",version="17.0.5+8",} 1.0
```

With this patch, we have both more meaningful values and a way to affect them if needed (e.g. vendor url etc.).

See with this patch:

```
jvm_info_total{runtime="Substrate VM",vendor="Oracle Corporation",version="GraalVM 23.0.0-dev71c7f0d86e29 Java 17 Mandrel Distribution",} 1.0
```
**EDIT:**
~(`version` field not showing the whole JDK version used is a bug in Mandrel build script, unrelated to this PR)~

Well, it seems reporting just the feature version was intentional. I'd like to change that, I think this makes more sense (added in a separate commit 96963cb01):

```
jvm_info_total{runtime="Substrate VM",vendor="Oracle Corporation",version="GraalVM 23.0.0-dev422cfafa4fd6 Java 17.0.4+8 Mandrel Distribution",} 1.0
```

### CODE REVIEW

**Review by galderz — APPROVED** [review]

**Comment by galderz** [comment]
@Karm style issues:

```
-        vendor = System.getProperty("org.graalvm.vendor","Oracle Corporation");
-        vendorUrl = System.getProperty("org.graalvm.vendorurl","https://www.graalvm.org/");
+        vendor = System.getProperty("org.graalvm.vendor", "Oracle Corporation");
+        vendorUrl = System.getProperty("org.graalvm.vendorurl", "https://www.graalvm.org/");
```

**Comment by Karm** [comment]
@galderz I think this is more on par with what HotSpot reports wrt to information granularity.   IMHO.

**Review by galderz — APPROVED** [review]

**Comment by galderz** [comment]
@Karm Yeah sure, no objections on my side.

**Comment by Karm** [comment]
@zakkak @galderz I thought this would be a 5 minutes patch, well... :) 

I was unable to use the native-image bash launcher script to affect these new org.graalvm.vendor, org.graalvm.vendorurl properties:

 ```
exec "${location}/../../../bin/java" -XX:+UnlockExperimentalVMOptions \
-XX:+EnableJVMCI -Dorg.graalvm.version="23.0.0-dev96963cb01ec9" -Dorg.graalvm.config="Mandrel Distribution" \
-Dorg.graalvm.vendor="KARM" -Dorg.graalvm.vendorurl="https://karms.biz" \
--upgrade-module-path "${location}/../../jvmci/graal.jar" --add-modules org.graalvm.truffle,org.graalvm.sdk \
--module-path "${location}/../../truffle/truffle-api.jar:${location}/../../jvmci/graal-sdk.jar" "${jvm_args[@]}" ${app_path_arg} "${cp_or_mp}" ${main_class} "${launcher_args[@]}"
```
    
until I added this [454f811](https://github.com/oracle/graal/pull/5355/commits/454f81138c2fe3cc89e8affb265a7758c3a63da9) in NativeImage.java.
 
It is configurable as expected now:

```
$ jfr print recording.jfr  | grep vendor -B2 -A2
jdk.InitialSystemProperty {
  startTime = 14:37:29.638
  key = "java.vm.vendor"
  value = "KARM"
}
--
jdk.InitialSystemProperty {
  startTime = 14:37:29.638
  key = "java.vendor.url"
  value = "https://karms.biz"
}

```

and Micrometer:

```
jvm_info_total{runtime="Substrate VM",vendor="KARM",version="GraalVM 23.0.0-dev96963cb01ec9 Java 17.0.5+8 Mandrel Distribution",} 1.0
```

**Comment by Karm** [comment]
Hello, @tkrodriguez @pejovica, may I ask for a review, please?

**Comment by tkrodriguez** [comment]
This isn't really my area.

**Comment by zakkak** [comment]
@oubidar-Abderrahim can you help us with assigning the right people to this PR? 

I suspect Doug Simon and/or Paul Woegerer would be suitable to review this, but I am not sure.

**Comment by tkrodriguez** [comment]
@christianwimmer could probably the most definitive answer about the contents of these properties.

**Comment by Karm** [comment]
Thanks, I looked in the git history of affected files and randomly selected some people who were not Doug/Paul/Christian in a poor attempt to load balance :-D  I'd be glad if @oubidar-Abderrahim could help.

Cheers

File: compiler/src/org.graalvm.compiler.serviceprovider/src/org/graalvm/compiler/serviceprovider/JavaVersionUtil.java
Comment (christianwimmer) [inline compiler/src/org.graalvm.compiler.serviceprovider/src/org/graalvm/compiler/serviceprovider/JavaVersionUtil.java:38]: There is no need to introduce this field. The `JavaVersionUItil` class is only for intended for numeric version checks.

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SystemPropertiesSupport.java
Comment (christianwimmer) [inline substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SystemPropertiesSupport.java:106]: The `runtime` property has nothing to do with Substrate VM as far as I know, because "runtime" means JDK (in contrast to "VM").

**Review by christianwimmer — CHANGES_REQUESTED** [review]

**Comment by Karm** [comment]
> Karm requested review from [christianwimmer](https://github.com/christianwimmer) and removed request for [olpaw](https://github.com/olpaw)

...did not intent to remove @olpaw :)

Ad GH Actions, I don't seem to have the right to re-run a single flow, but I think it's worth just re-running:

```
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by
 'NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f3bd8da45e0>: 
Failed to establish a new connection: [Errno 101] Network is unreachable')': /packages/e9/59/43fc36c5ee316bb9aeb7cf5329cdbdca89e5749c34d5602753827c0aa2dc/pylint-2.4.4-py3-none-any.whl
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by
 'NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0x7f3bd8da4730>:
Failed to establish a new connection: [Errno 101] Network is unreachable')': /packages/e9/59
```

File: compiler/src/org.graalvm.compiler.serviceprovider/src/org/graalvm/compiler/serviceprovider/JavaVersionUtil.java
Comment (olpaw) [inline compiler/src/org.graalvm.compiler.serviceprovider/src/org/graalvm/compiler/serviceprovider/JavaVersionUtil.java]: @dougxc are you ok with removing `JAVA_VERSION` from `JavaVersionUtil`?

File: substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/VM.java
Comment (olpaw) [inline substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/VM.java:49]: Wouldn't `"Unknown JDK Runtime Environment"` be more appropriate?
If `java.runtime.name` is not found we might run on something exotic. So let's not pretend.

Comment (Karm) [reply substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/VM.java]: Ack.

**Review by olpaw — APPROVED** [review]

**Comment by Karm** [comment]
Hello, @oubidar-Abderrahim can this patch be integrated now? Are there any further comments?

**Comment by Karm** [comment]
Hello @christianwimmer , can this patch be integrated now? Are there any further comments?

**Comment by oubidar-Abderrahim** [comment]
This PR was mirrored to bitbucket and is currently pending approval. Once the mirrored PR is merged, this one will be as well.
Tracked on GR 42356.

**Comment by Karm** [comment]
Hello @oubidar-Abderrahim, @olpaw ,

This is a trivial PR about populating system properties related to JVM runtime and vendor naming.
It has been opened for almost a month now.
There was a message about it being mirrored to Bitbucket and tracked as [GR-42356].
The PR created for that, #5500, was wrong, taking an old version before the last review and updates.
The erroneous PR was closed later.

I would like to kindly ask for this PR, #5355, to be merged. It was thoroughly reviewed by several parties, updated and acked.

**Comment by oubidar-Abderrahim** [comment]
@Karm The issue with #5500 was that I started the integration process before you pushed the last update. To mitigate this type of issue in the future, any PR should have at least 1 approval from an Oracle member before starting the integration process, and once it starts, users should not push any more updates.
For this PR, the merge process started and the PR is in the queue now, it should be merged soon.

**Comment by Karm** [comment]
@oubidar-Abderrahim Thank you for looking into it :+1:

**Pull request merged**
